File: api/note/src/main/java/com/erinbro/librimem/note/model/Note.java
package com.erinbro.librimem.note.model;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.util.Date;

@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity(name = "Note")
@Table(name = "note")
public class Note implements INote{
    @Id
    @GeneratedValue(
            strategy = GenerationType.IDENTITY
    )
    private Integer id;

    private String type;
    @Column(name = "entity_id", nullable = false)
    private Integer entityId;
    @Column(name = "chapter_id", nullable = false)
    private Integer chapterId;

    private boolean favorite;

    @Column(nullable = false)
    private String title;

    /**
     * The actual note
     */
    @Column(nullable = false)
    private String note;

    @CreatedDate
    @Column(name = "created_at")
    private Date createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private Date updatedAt;
}

File: frontend/apps/librimem/src/app/models/Note.ts
import { INote } from '@librimem/api-interfaces';

export class Note implements Omit<INote, "id">{
  userId = 0;
  title = "";
  note = "";
  type = "CHAPTER";
  entityId = 0;
  chapterId = 0;
  favorite = false;
  createdAt = new Date().toJSON();
  updatedAt = new Date().toJSON();
}
